### PR TITLE
Rate brave dp

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves : https://github.com/brave/brave-browser/issues/8243

## Submitter Checklist:

- [X] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [X] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [X] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Please follow the spec https://docs.google.com/document/d/1MHMXzlF7vvXQlISet0Pq4DYfxxMbOWSNjxrTQhW3trM/edit to test different scenarios. 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/rate/RateDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+import org.chromium.chrome.browser.util.ConfigurationUtils;
+import org.chromium.chrome.browser.rate.RateFeedbackUtils;
+import org.chromium.chrome.browser.night_mode.GlobalNightModeStateProviderHolder;
+
+public class RateDialogFragment extends DialogFragment implements View.OnClickListener {
+    private static final String SAD = "sad";
+    private static final String NEUTRAL = "neutral";
+
+    private String mUserSelection;
+	private boolean mIsFeedbackShown;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rate/RateDialogFragment.java:44]: @deeppandya could you please align everything in that file as it looks weird.

File: android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+public class RateFeedbackUtils {
+	private static final String TAG = "Rate";
+
+	public interface RateFeedbackCallback {
+		void rateFeedbackSubmitted();
+	}
+
+	public static class RateFeedbackWorkerTask extends AsyncTask<Void> {
+		private String mUserSelection;
+	    private String mUserFeedback;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java:41]: alignment problem

File: android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+	        mCallback.rateFeedbackSubmitted();
+	    }
+	}
+
+	private static void sendRateFeedback(String userSelection, String userFeedback) {
+		Context context = ContextUtils.getApplicationContext();
+		String appVersion = AboutChromeSettings.getApplicationVersion(context, AboutSettingsBridge.getApplicationVersion());
+		StringBuilder sb = new StringBuilder(); 
+		String http = "https://laptop-updates.brave.com/1/feedback";
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java:70]: do the hardcode on top of the class as final static for easy modifications in the future.

File: android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		try {  
+		    URL url = new URL(http);  
+		    urlConnection = (HttpURLConnection) url.openConnection();
+		    urlConnection.setDoOutput(true);   
+		    urlConnection.setRequestMethod("POST");  
+		    urlConnection.setUseCaches(false);  
+		    urlConnection.setRequestProperty("Content-Type","application/json");  
+		    urlConnection.connect();
+
+            JSONObject jsonParam = new JSONObject();
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rate/RateFeedbackUtils.java:85]: alignment

File: android/java/org/chromium/chrome/browser/rate/RateUtils.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        return mSharedPreferences.getInt(PREF_NEXT_APP_OPEN_COUNT, 0);
+    }
+
+    public void setPrefNextAppOpenCount() {
+        int currentAppOpenCount = getPrefNextAppOpenCount();
+        int nextAppOpenCount;
+
+        if (currentAppOpenCount == 0) {
+            nextAppOpenCount = APP_OPEN_14;
+        } else if (currentAppOpenCount == APP_OPEN_14) {
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/rate/RateUtils.java:91]: shouldn't we do here `if (currentAppOpenCount >= APP_OPEN_14 || currentAppOpenCount <= APP_OPEN_41)`. The reason for that is that if for some reason(crash or whenever) we don't reach that code on `APP_OPEN_14` we will sit on old value forever.

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/rate/RateUtils.java:91]: @SergeyZhukovsky i agree with you. i am updating the code.

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/rate/RateUtils.java:91]: @SergeyZhukovsky we update this value only when we see the rate dialog for next time occurrence.

**Pull request merged**